### PR TITLE
Update `Checkbox` tests for M2/M3

### PR DESCRIPTION
**Pull request by QuncCccccc**
Updated unit tests for `Checkbox` to have M2 and M3 versions.

More info in #127064

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/test/material/checkbox_test.dart [hunk 1]
```diff
@@ -504,24 +564,20 @@ void main() {
...
-    final bool material3 = themeData.useMaterial3;
-    final ColorScheme colorScheme = material3
-        ? const ColorScheme.light().copyWith(primary: activeColor)
-        : const ColorScheme.light().copyWith(secondary: activeColor);
-    themeData = themeData.copyWith(colorScheme: colorScheme);
+    final ColorScheme colorScheme = const ColorScheme.light().copyWith(secondary: activeColor);
+    theme = theme.copyWith(colorScheme: colorScheme);
     await tester.pumpWidget(buildFrame(
-      themeData: themeData),
+      themeData: theme),
```
Comment (TahaTesser) [inline packages/flutter/test/material/checkbox_test.dart:580]: Nice

File: packages/flutter/test/material/checkbox_test.dart [hunk 2]
```diff
@@ -1624,7 +1964,7 @@ void main() {
     expect(find.text(tapTooltip), findsOneWidget);
   });
 
-  testWidgets('Checkbox has default error color when isError is set to true - M3', (WidgetTester tester) async {
+  testWidgets('Checkbox has default error color when isError is set to true - M3 only', (WidgetTester tester) async {
```
Comment (TahaTesser) [inline packages/flutter/test/material/checkbox_test.dart:1967]: There is `Material3` in other tests but here its `M3` for the "only" test. Is this intentional?

File: packages/flutter/test/material/checkbox_test.dart [hunk 3]
```diff
@@ -1841,7 +2181,62 @@ void main() {
...
+      );
+    }
+
+    RenderBox getCheckboxRenderer() {
+      return tester.renderObject<RenderBox>(find.byType(Checkbox));
+    }
+
+    // Checkbox is unselected, so the default BorderSide appears and fillColor is checkbox's background color.
+
+    await tester.pumpWidget(buildApp());
```
Comment (TahaTesser) [inline packages/flutter/test/material/checkbox_test.dart:2215]:
```suggestion
    // Checkbox is unselected, so the default BorderSide appears and fillColor is checkbox's background color.
    await tester.pumpWidget(buildApp());
```

If this comment is related

**Review by TahaTesser — CHANGES_REQUESTED** [review]

**Comment by HansMuller** [comment]
As you update tests, don't forget to also update https://docs.google.com/spreadsheets/d/1TEmxXloSu0cHNQuBn6xjH895-lnSOqH3pln6JVACorc/edit#gid=73177703 per "Step 3" in https://github.com/flutter/flutter/issues/127064

**Review by TahaTesser — APPROVED** [review]
LGTM!

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**